packages/design-system/src/components/Icons/SvgIcon.tsx

```diff
@@ -76,7 +75,7 @@ export function SvgIcon({
       aria-hidden={ariaHidden}
       className={svgClasses}
       focusable={false}
-      id={iconId}
+      id={isSrVisible && rootId}
```

Comment:
I actually think this logic needs to be `id ?? isSrVisible && rootId` because if the user explicitly wants an id, we need to give it to them. They're probably using it for something

Reply:
good point!

Reply:
i'm finding errors in our unit tests using this syntax. 
changing it to `id={ id ?? isSrVisible ? rootId : undefined }` _seems_ to work - thoughts?

Reply:
Yep, that is better, more correct